### PR TITLE
[BLOG] Pydata/Sparse: Maintenance and docs overhaul

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
## Text styling

- [ ] The blog is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags in order to do so (with only one level-one header).
- [ ] All links describe where they link to (for example, check the [Quansight labs website](https://labs.quansight.org/)).
- [ ] Any kind of styling that the author uses (for example, bold for emphasis) is consistent throughout the blog.

## Non-text contents

- [ ] Blog post featured image is in PNG or JPEG format, **not** SVG.
- [ ] All content is represented as text (for example, images need alt text and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing gifs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If it were to be read as plain text, the blog still makes sense and no information is missing.

@hameerabbasi

### CODE REVIEW

**Review by hameerabbasi — COMMENTED** [review]
Looks great overall, Dea, thanks for writing this up. I especially enjoyed your take on this journey. I've made some accuracy suggestions; and two optional content suggestions.

File: apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+Regarding the code of the benchmarks, `CodSpeed` works with `pytest` as I mentioned. So, we can use parametrized fixtures. The fixture `benchmark` runs `CodSpeed`.
+
+We are now benchmarking matmul, elemwise and tensordot with `CodSpeed`. We are also comparing elemwise when using `Numba` vs. using `Finch`.
+
+## Acknowledgements
+
+I would like to thank:
+
+Hameer Abbas who was my guide during this time.
+Mateus Sokól, who also helped me very kindly in the beginning of the project.
```
[the remark this answers is not in the document]
Comment (DeaMariaLeon) [reply apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md:169]: Sorry about that!

File: apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+The Release Drafter is a GitHub action that can be found in "Marketplace Actions". It must be added to the repository with a YAML workflow file. It is stored in ` .github/workflows/release-drafter.yml`. After that, it needs to be configured with another YAML file, now with the name `.github/release-drafter.yml`.
+
+Both files need to be in the `main` branch already in order to start working. Because of this, it is easier to add these files directly to the repository on GitHub than with a pull request. But that can be done only by the owners of the repositories. When someone who is not the owner of the repository creates the pull request to add them, the workflow fails. The pull request must be merged like that. Once the files are in the default branch, the Release Drafter starts working.
+
+With all this, a document that is the release draft keeps updating itself automaticaly when pull requests are merged. I included the option to categorize pull requests and it does it by setting labels to the pull requests, depending on their titles. For this, I added the "Autolabeler" funcionality.
+
+### Improved and expanded benchmarks
```
[the remark this answers is not in the document]
Comment (DeaMariaLeon) [reply apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md:142]: The link to CodSpeed docs is on line 149, when I start writing about this library. Should I move things around?

[the remark this answers is not in the document]
Comment (hameerabbasi) [reply apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md:142]: I just mean that linking docs to things you experienced might be useful.

[the remark this answers is not in the document]
Comment (hameerabbasi) [reply apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md:142]: Specific pages in the docs, I mean.

File: apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
+---
+
+title: "Pydata sparse: Maintenance and docs overhaul"
```
[the remark this answers is not in the document]
Comment (DeaMariaLeon) [reply apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md]: I tried `sparse` but couldn't make it work, so I took the second option.

File: apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+---
+
+Hi, I'm [Dea María Léon](https://github.com/DeaMariaLeon). I had the pleasure to work at Quansight Labs for three months. The objective was to perform maintenance tasks and a documentation overhaul to the `Pydata.sparse` project.
+
+In this post, I'll describe the work during that time. From now on, I'll refer to the project only as `sparse`.
+
+---
+
+- [Introduction to pydata sparse](#introduction-to-pydata-sparse)
```
[the remark this answers is not in the document]
Comment (DeaMariaLeon) [reply apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md]: Done

File: apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+
+I was given "Admin" rights to `sparse` in the host (`Read the Docs`), and I activated the "trigger builds from pull requests". This is the feature that lets us preview the documentation builds with each commit in `GitHub` with a link.
+
+The tricky part was that Hameer, the owner of the project on `GitHub`, also needed to link his `Read the Docs` account to `GitHub`.
+
+If I had started by deploying `sparse` directly, without installing my own fork first, it would have been impossible to find the missing part. The reason being that I didn't have access to all the project settings on `GitHub`. But I knew what was needed: to connect both accounts (`GitHub` and `Read the Docs`). This part was a good troubleshooting exercise.
+
+At the time of writing, the new website is available only under the "latest" version on `Read the Docs`. The stable version is still the `Sphinx` website, as the next release is still not ready. But it will be part of the next release.
+
+The resulting documentation site can be seen [here.](https://sparse-nd--747.org.readthedocs.build/en/latest/)
```
Comment (hameerabbasi) [inline apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md:108]: Seems like you corrected the last part of the URL but forgot the first part. 😅 

```suggestion
The resulting documentation site can be seen [here.](https://sparse.pydata.org/en/latest/)
```

File: apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+- [Acknowledges](#acknowledges)
+
+- [References](#references)
+
+---
+
+## Introduction to pydata sparse
+
+`sparse` is a library that implements operations for sparse arrays with two or more dimensions.
```
[the remark this answers is not in the document]
Comment (DeaMariaLeon) [reply apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md]: True!!

File: apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (hameerabbasi) [reply apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md]: > Sorry about that!

No worries. 😄

**Comment by hameerabbasi** [comment]
> Any ideas on why this is not working? Screenshot 2024-09-04 at 10 15 34

Seems like there's no avatar for your author slug, @melissawm is there something we should do? Perhaps all interns will face this?

File: apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md (hunk 5 above)
Comment (DeaMariaLeon) [reply apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md]: Yes I saw it too late!

File: apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (DeaMariaLeon) [reply apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md:142]: I added more links as you suggested. Thanks for that.

File: apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+
+With these principles I divided the original content to match these guidelines. I also added the section for "contributing" and a specific section for the introduction.
+
+### Added missing docstrings
+
+The final requirement related to the website was to add the missing docstrings so the API is fully documented.
+
+## Maintenance tasks
+
+### Added community tools to repository on GitHub
```
[the remark this answers is not in the document]
Comment (DeaMariaLeon) [reply apps/labs/posts/pydata-sparse-maintenance-and-docs-overhaul.md:126]: Thanks again. :)

**Review by hameerabbasi — APPROVED** [review]
LGTM, thanks for the blog post! Just waiting on Melissa's reply for your author image.

**Comment by DeaMariaLeon** [comment]
I need to fix a link to the same document that is not working. Sorry.

**Comment by pavithraes** [comment]
> Seems like there's no avatar for your author slug

Dea, please submit a picture in the form to add you to the Labs team page ([pinged on Slack too](https://quansight.slack.com/archives/C079P4SSVNX/p1725961555652669?thread_ts=1721211654.636809&cid=C079P4SSVNX)) -- we can then update your profile with it & it'll be reflected in the published post.

**Comment by DeaMariaLeon** [comment]
I just did it. Thanks again.

**Comment by pavithraes** [comment]
I see your profile picture now: https://labs-git-fork-deamarialeon-blog-quansight.vercel.app/blog/pydata-sparse-maintenance-and-docs-overhaul :)

**Pull request merged**